popde/density_estimate.py

```diff
@@ -1,3 +1,97 @@
+class SimpleKernelDensityEstimation:
+    """
+    Fit and evaluate general Kernel Density Estimation (KDE) 
+
+    Methods:
+    --------
+    fit(data, bandwidth=None, kernel=None):
+        Fit the KDE to new data.
+
+    evaluate(points):
+        Evaluate the KDE at given data points.
+
+    Examples:
+    ---------
+    # Create a KDE instance and fit it to data
+    data = np.random.randn(100)
+    kde = SimpleGaussianKernelDensityEstimation(data)
+    kde.fit()
+
+    # Evaluate the KDE at new data points
+    new_data = np.linspace(-3, 3, 100)
+    density_values = kde.evaluate(new_data)
+    """
+
+    def __init__(self, data, bandwidth=1.0, kernel=['gaussian'], dim_names=None):
+        """
+        Initialize the KernelDensityEstimation object.
+
+        Parameters:
+        -----------
+        data : array-like
+            The input data for which KDE will be estimated.
+
+        bandwidth : float, optional (default=1.0)
+            Bandwidth parameter controlling the width of the kernels.
+            Larger bandwidths lead to smoother density estimates.
+
+        kernel : list of str, optional (default=['gaussian'])
+            The kernel function used for smoothing.
+
+        dim_names : list of the dimension names (m1,z, chi_eff) 
+             If dim_names is not Nonecheck that the dimensionality
+             of the data array matches with len(dim_names)
+        """
+        import numpy as np
+
+        self.data = np.asarray(data)
+        self.bandwidth = bandwidth
+        if self.kernel in kernel:
+            self.kernel = kernel[0]#may need to fix this
+        else:
+            raise NotImplementedError("Unsupported kernel. Supported kernels: 'gaussian',")
+
+        self.dim_names = dim_names
+        if dim_names is not None:
+            self.check_dimensionality()
+
+
+   def check_dimensionality(self):
+        if len(self.data.shape) != len(self.dim_names):
```

Comment:
have you checked this?  ie what is `len(self.data.shape)` ?
For example if you are using 3d data, what should `self.data.shape` be ?

Reply:
I checked and if I have a data in shape like (10, 10, 10) , the len will be 3. I am not sure if we can have data shape (10, 10, 10, 1)

Reply:
fixed it and seems to work on an example